doc/manual/src/glossary.md

```diff
@@ -208,44 +208,85 @@
 
-    A program that is produced by a [derivation] is built with a [build system] that can be executed by the [derivation system](language/derivations.md#attr-system).
-    After it has been built, the [build system] is no longer relevant.
+  - [system type]:{#gloss-system-type}
 
-  - [host system]{#gloss-host-system}
+    An identifier for a specific [platform] or (in rare cases) a class of compatible platforms, such as `x86_64-v3-linux`.
 
-    [host system]: #gloss-host-system
+    To build a derivation locally, the [derivation `system` field](language/derivations.md#attr-system) must occur in the local Nix installation's [`system`](command-ref/conf-file.md#conf-system) or [`extra-platforms`](command-ref/conf-file.md#conf-extra-platforms) settings, among other rules.
+    Otherwise, a [remote build](advanced-topics/distributed-builds.md) is needed.
 
-    A representation of the processor architecture and platform where build products will run on, typically a dash-separated string.
+  - [build platform]{#gloss-build-platform}
 
-    This is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+    [build platform]: #gloss-build-platform
 
-    Programs that run natively in a [derivation] must be configured with a [host system] that can by executed by the [derivation system](language/derivations.md#attr-system).
+    The [platform] that performs the build or compilation, typically a dash-separated string.
 
-  - [target system]{#gloss-target-system}
+    A program that is produced by a [derivation] has a [build configuration] with a [build platform] that can be executed by the [derivation system](language/derivations.md#attr-system).
+    After it has been built, the [build platform] is no longer relevant.
 
-    [target system]: #gloss-target-system
+    This term is based on the [GCC system terminology](#gloss-gcc-system-terminology).
 
-    A representation of the processor architecture and platform that a compiler will produce, typically a dash-separated string.
+  - [host platform]{#gloss-host-platform}
 
-    When discussing a build that does not produce a compiler, we only use [build system] and [host system] to describe it; we try to avoid discussing the actual build from the perspective its compiler's configuration.
+    [host platform]: #gloss-host-platform
 
-    This is based on the [GCC system terminology](#gloss-gcc-system-terminology).
+    The [platform] where build products will run on, typically a dash-separated string.
 
-    If the output of a compiler is to be run in a [derivation], the compiler's [target system] must be executable by the [derivation system](language/derivations.md#attr-system).
+    Programs that run natively in a [derivation] must have a [build configuration] with a [host platform] that can by executed by the [derivation system](language/derivations.md#attr-system).
```

Comment:
```suggestion
    Programs that run when a [derivation] is built must have <as before>
```

Not sure here. Apparently you're using "in a derivation" with the meaning "while running the `builder` executable of a derivation`, which is often abbreviated by "when building a derivation", and I don't think it's semantically wrong, but given that *derivation* is such a mystical term for beginners, we should be careful with condensing too much. I'd rather cut down on "building" a derivation or at least define what that *actually* means in the glossary as well. Actually this is something that I wanted to do some time ago but got rabbit-holed with.

Reply:
I think while the build platform is only a property of the build configuration, the host and target platforms are properties of the built program itself too. I think it makes sense to break symmetry and stop talking about the build configuration for these latter too to further emphasize what information is relevant when.